File: project_euler/function_library/discarded/root_finder_guess.py
from numpy import *;
import time;

def eps(): return spacing(single(1))

def root_finder_guess(f,guess,dx=eps(),precision=eps()):
    start=time.time(); 
    slope=(f(guess+dx)-f(guess))/dx    
    guess-=(f(guess)/slope)
    while abs(f(guess))>=precision:
        slope=(f(guess+dx)-f(guess))/dx
        guess-=(f(guess)/slope)
    return guess,time.time()-start

File: project_euler/function_library/root_finder_guess.m
function [a t]=root_finder_guess(f,guess,dx,precision)
t1=now;
if nargin==3
	precision=eps;
end
while abs(f(guess))>=precision
	slope=(f(guess+dx)-f(guess))/dx;
	guess=guess-(f(guess)/slope);
end
a=guess;
t=run_time(t1);
end